zigzag/parser/onnx/GemmParser.py

```diff
@@ -29,58 +29,88 @@ def run(self) -> LayerNode:
         """! Run the parser"""
         return self.generate_layer_node()
 
+    def get_operand_source_user_format(self, predecessors: list[int]):
+        """Set input source and indicate constat operands"""
+        match len(predecessors):
+            case 0:
+                # No source operands -> assume one is constant
+                return {"W": self.node_id}
+            case 1:
+                # One source operand, one constant
+                return {"W": self.node_id, "I": predecessors[0]}
+            case 2:
+                # Two source operands, none are constant (W and I can be swapped)
+                return {"W": predecessors[1], "I": predecessors[0]}
+            case _:
+                raise ValueError("No more than 2 layer predecessors expected")
+
     def get_layer_node_user_format(
         self,
-        batch_size: int,
-        size_c: int,
-        size_k: int,
-        size_d: int,
+        input_shape: list[int],
+        output_shape: list[int],
     ) -> dict[str, Any]:
         """! Generate layer data in user input format for MatMul or GEMM ONNX nodes.
-        I[B][K][C} * W[1 or B][C][D]-> O [B][K][D]
-        """
-
-        data: dict[str, Any] = {}
-        data["id"] = self.node_id
-        data["name"] = self.node.name
-        data["operator_type"] = self.node.op_type
-        data["loop_dims"] = ["B", "C", "K", "D"]
-        data["loop_sizes"] = [batch_size, size_c, size_k, size_d]
-        data["dimension_relations"] = []
+        I[B]   [K][C] * W([B])   [C][D]-> O [B]   [K][D] or
+        I[B][H][K][C] * W([B][H])[C][D]-> O [B][H][K][D]
```

Comment:
K is typically used for output channels and is not present for the input operand. Is it possible input should have C and D, weight should have C and K and output has K and D?